skbio/diversity/_util.py

```diff
@@ -22,9 +22,6 @@ def _validate_counts_vector(counts, suppress_cast=False):
     """
     counts = np.asarray(counts)
 
-    if not suppress_cast:
-        counts = counts.astype(int, casting='safe', copy=False)
```

Comment:
would it be possible to test the data type appear to be numeric? It would be great to get a useful error message when passing in a string or vector of dict fo instance.

Reply:
👍